packages/astro/config.mjs

```diff
@@ -1,3 +1,43 @@
 export function defineConfig(config) {
 	return config;
 }
+
+export function getViteConfig(inlineConfig) {
+	// Return an async Vite config getter which exposes a resolved `mode` and `command`
+	return async ({ mode, command }) => {
+		// Vite `command` is `serve | build`, but Astro uses `dev | build`
+		const cmd = command === 'serve' ? 'dev' : command;
+
+		// Use dynamic import to avoid pulling in deps unless used
+		const [
+			{ mergeConfig },
+			{ nodeLogDestination },
+			{ openConfig },
+			{ createVite },
+			{ runHookConfigSetup, runHookConfigDone },
+		] = await Promise.all([
+			import('vite'),
```

Comment:
why are these dynamic imports?

Reply:
This avoids pulling in the dependencies unless `getViteConfig` is actually called. This was a bit defensive but it avoids loading all these dependencies through Proload/Vite for an `astro.config.ts` file that only uses `import { defineConfig } from "astro/config"`.

Reply:
Aren't those dependencies loaded anyways? if you are loading an astro config then that means you are loading `astro dev` or `astro build`, no?

Reply:
Yes but all the `astro` deps are being loaded through Node, whereas Proload/Vite might run all the file through a transform step before loading them in Node (and thus all the `import` statements). I didn't benchmark this, but my guess was that it would speed up loading `astro.config.ts` files since the dynamic deps wouldn't also need to get transformed by Proload/Vite.

Reply:
ah, ok i see. that definitely is unfortunate. but it makes sense.